File: src/main/kotlin/com/xs0/gqlktx/schema/builder/TypeKind.java
package com.xs0.gqlktx.schema.builder;

import com.xs0.gqlktx.ann.*;

import java.lang.annotation.Annotation;

@GraphQLEnum("__TypeKind")
public enum TypeKind {
    SCALAR(GraphQLScalar.class),
    ENUM(GraphQLEnum.class),

    INPUT_OBJECT(GraphQLInput.class),

    INTERFACE(GraphQLInterface.class),
    OBJECT(GraphQLObject.class),
    UNION(GraphQLUnion.class),

    NON_NULL(null),
    LIST(null);

    private final Class<? extends Annotation> explicitAnnotation;

    TypeKind(Class<? extends Annotation> explicitAnnotation) {
        this.explicitAnnotation = explicitAnnotation;
    }

    public Class<? extends Annotation> getExplicitAnnotation() {
        return explicitAnnotation;
    }
}

File: src/main/kotlin/com/xs0/gqlktx/schema/builder/TypeKind.kt
package com.xs0.gqlktx.schema.builder

import com.xs0.gqlktx.*
import kotlin.reflect.KClass

@GqlEnum("__TypeKind")
enum class TypeKind(val explicitAnnotation: KClass<out Annotation>?) {
    SCALAR(GqlScalar::class),
    ENUM(GqlEnum::class),

    INPUT_OBJECT(GqlInput::class),

    INTERFACE(GqlInterface::class),
    OBJECT(GqlObject::class),
    UNION(GqlUnion::class),

    NON_NULL(null),
    LIST(null)
}
